File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        string_x = str(x)
        sign = ""
        string_rst = ""

        for index in range(len(string_x)):
            if(index == 0 and string_x[0] == "-"):
                sign = string_x[0]
                continue
            
            # always append to the front
            string_rst = string_x[index] + string_rst

        if (sign == "-"):
            string_rst = sign + string_rst

        # computing the result
        if (-2**31 <= int(string_rst) <= 2**31 - 1):
            rst = int(string_rst)
        else:
            rst = 0

        return rst

            


        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        string string_x = to_string(x);
        string sign = "";
        string string_rst = "";

        for (int index = 0; index<string_x.size(); index++){
            if (index == 0 && string_x[index] == '-'){
                sign = string_x[0];
                continue;
            }

            string_rst = string_x[index] + string_rst;
        }

        if (sign == "-"){
            string_rst = sign + string_rst;
        }

        long long value = stoll(string_rst);
        return (-2147483648LL <= value && value <= 2147483647LL) ? int(value) : 0;
    }
};